data/ai/micro_ais/cas/ca_bloodlust.lua

```diff
@@ -0,0 +1,132 @@
+local AH = wesnoth.require "ai/lua/ai_helper.lua"
+local MAIUV = wesnoth.require "ai/micro_ais/micro_ai_unit_variables.lua"
+local M = wesnoth.map
+
+local function bloodlust_unit_attack_weakest_adj_enemy(ai, bloodlust_unit)
+    -- Attack the enemy with the fewest hitpoints adjacent to 'bloodlust_unit', if there is one
+    -- Returns status of the attack:
+    --   'attacked': if a unit was attacked
+    --   'killed': if a unit was killed
+    --   'no_attack': if no unit was attacked
+
+    if (bloodlust_unit.attacks_left == 0) then return 'no_attack' end
+
+    local min_hp, target = math.huge, nil
+    for xa,ya in wesnoth.current.map:iter_adjacent(bloodlust_unit) do
+        local enemy = wesnoth.units.get(xa, ya)
+        if AH.is_attackable_enemy(enemy) then
+            if (enemy.hitpoints < min_hp) then
+                min_hp, target = enemy.hitpoints, enemy
+            end
+        end
+    end
+
+    if target then
+        AH.checked_attack(ai, bloodlust_unit, target)
+        if target and target.valid then
+            return 'attacked'
+        else
+            return 'killed'
+        end
+    end
+
+    return 'no_attack'
+end
+
+local function get_level_bloodlust_unit(cfg)
+    local filter = wml.get_child(cfg, "filter") or { id = cfg.id }
+    local bloodlust_unit = AH.get_units_with_moves {
+        side = wesnoth.current.side,
+        { "and", filter }
+    }[1]
+    return bloodlust_unit
+end
+
+local ca_bloodlust = {}
+
+function ca_bloodlust:evaluation(cfg)
+    if get_level_bloodlust_unit(cfg) then return cfg.ca_score end
+    return 0
+end
+
+function ca_bloodlust:execution(cfg)
+    -- the bloodlust unit does a random wander in area given by @cfg.wandering_ground until it finds
+    -- and kills an enemy unit and then rinse and repeat
+    -- note most of this is just some edits of the hunter Micro AI
+    -- but with no home/returning things
+    -- the bloodlust_unit kills and eliminates anything in the defined arena
+
+    local bloodlust_unit = get_level_bloodlust_unit(cfg)
+    local bloodlust_unit_vars = MAIUV.get_mai_unit_variables(bloodlust_unit, cfg.ai_id)
+
+    -- If bloodlust_unit_status is not set for the bloodlust_unit -> default behavior -> random wander
+    if (not bloodlust_unit_vars.bloodlust_unit_status) then
```

Comment:
I've only just realised that `bloodlust_unit_status` is only set to `nil`. Unlike the Hunter CA's `hunting_status`, this variable doesn't seem to have a purpose.

It might be counting as a change for the purpose of checking whether the execution function has changed the game-state, but if it is then that would likely be a bug.